### DISCUSSION TITLE
'notFound' template not return current user query

### DISCUSSION THREAD

**Discussion by AlonsoK28** [digitalfortress-tech/typeahead-standalone#44] [Q&A] [answered]
This is usefull to render templates with an improved UX

example
```js
const template = `
          <div class="d-flex align-items-center justify-content-center">
            <span>Not found results founds with your query {{query}}</span>
          </div>`;
```

More about this into `jQuery Typeahead` documentation http://www.runningcoder.org/jquerytypeahead/documentation/
![image](https://user-images.githubusercontent.com/37458700/212781890-4895906f-a3d1-4317-bf80-631bc09bc901.png)

**Comment by SneakyGerald** [comment]
I think it would be useful to have more data in all templates: header, footer and notFound, to give more information to the user for better UX, for example, total results found, manage additional logic to show "see more" button depending on results found, etc...

**Reply by niketpathak** [reply]
Hello, yes, thank you for your valuable suggestions @AlonsoK28 and @SneakyGerald . This will be coming up soon in the next release. Will update when that happens! Good day

**Comment by AlonsoK28** [comment, marked as answer]
for me a workaround was like the following:

```js
notFound: () => {
        const query = document.querySelector('#searchInputTypeahead').value;
        const slug = gen(generateSlugSearch(query), slugTypes.search);
        const template = `
          <div class="d-flex flex-column align-items-center justify-content-center">
            <span>Aun no hemos encontrado resultados</span>
            <button class="btn btn-primary my-2" type="submit" onclick="window.location.href = '${slug}'">
                <div class="d-flex align-items-center">
                    <i class="fas fa-search pe-1"></i>
                    Ampliar más la búsqueda
                </div>
            </button>
          </div>
        `;
        return template;
      },
```

**Reply by niketpathak** [reply]
Hello @AlonsoK28 , @SneakyGerald , as of [v4.16.0](https://github.com/digitalfortress-tech/typeahead-standalone/releases/tag/v4.16.0), typehead provides additional data within templates. You may refer the [docs](https://typeahead.digitalfortress.tech/#styles?id=templates) for implementation details and a live example can be found on the [playground](https://typeahead.digitalfortress.tech/#playground?id=colors-example)

@AlonsoK28,

To access the query within the notFound template, you can simply do
```js
notFound: (resultSet) => {
   return `Nothing found for "${resultSet.query}"`;
}
```

Besides the `query`, you can also access additional data like the `count` (i.e. total suggestions found) and so on. 

Hope that helps!
